server/service/hosts.go

```diff
@@ -186,18 +196,44 @@ func deleteHostsEndpoint(ctx context.Context, request interface{}, svc fleet.Ser
 		StatusFilter: req.Filters.Status,
 		TeamFilter:   req.Filters.TeamID,
 	}
-	err := svc.DeleteHosts(ctx, req.IDs, listOpt, req.Filters.LabelID)
-	if err != nil {
-		return deleteHostsResponse{Err: err}, nil
+
+	// Since bulk deletes can take a long time, after DeleteHostsTimeout, we will return a 202 (Accepted) status code
+	// and allow the delete operation to proceed.
+	var err error
+	deleteDone := make(chan bool, 1)
+	ctx = context.WithoutCancel(ctx) // to make sure DB operations don't get killed after we return a 202
+	go func() {
+		err = svc.DeleteHosts(ctx, req.IDs, listOpt, req.Filters.LabelID)
```

Comment:
Any issues with using `ctx` here? (in case the timeout is hit and the request is finished)

Reply:
I don't see an issue with request and timeout finishing at the same time -- either a 200 or a 202 will be returned. It doesn't really matter.

But there is another side effect due to me using `context.WithoutCancel(ctx)`. The delete will continue even if a client disconnects on their end. I assume this is fine -- this is a corner case that we don't test anyway.